website/docs/d/cloudant.html.markdown

```diff
@@ -0,0 +1,61 @@
+---
+layout: "ibm"
+page_title: "IBM : ibm_cloudant"
+description: |-
+  Get information about Cloudant instance.
+subcategory: "Cloudant"
+---
+
+# ibm_cloudant
+
+Provides a read-only data source for an existing IBM Cloud Cloudant service. You can then reference the fields of the data source in other resources within the same configuration using interpolation syntax.
+
+## Example Usage
+
+```hcl
+resource "ibm_resource_instance" "cloudant" {
+ 	  name     = "cloudant-service-name"
+ 	  service  = "cloudantnosqldb"
+ 	  plan     = "lite"
+ 	  location = "us-south"
+ }
+
+ data "ibm_cloudant" "instance" {
+ 	  name     = ibm_resource_instance.cloudant.name
+ }
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` (Required, String) The name of the IBM Cloudant resource instance.
+* `location` (Optional, String) The location or the environment in which instance exists.
```

Comment:
In this case, I think there is no `default` value. In a `data` block certain attributes should be defined and based on that, the existing instance can be filtered.
So if this is not defined, it will not search based on `location` value.
>A data block requests that Terraform read from a given data source and export the result under the given local name.
source: https://www.terraform.io/docs/language/data-sources/index.html

Reply:
Oh, I see, thanks. I hadn't realise this constraint applied only [after listing resources](https://github.com/cloudant/terraform-provider-ibm/blob/e94ae1aa0ca31fd1c45cb0ab071300dc30410ced/ibm/data_source_ibm_resource_instance.go#L180) - I thought it was part of the selector used to list resources like the resource group.